lib/dns.js

```diff
@@ -263,26 +274,27 @@ function resolver(bindingName) {
     req.hostname = name;
     req.oncomplete = onresolve;
     req.ttl = !!(options && options.ttl);
-    var err = binding(req, name);
+    var err = this._handle[bindingName](req, name);
     if (err) throw errnoException(err, bindingName);
     return req;
   };
```

Comment:
Perhaps make the function anonymous by default then set the name to `bindingName` before returning...

```js
const fn = (name, /* options, */ callback) => {
  /* ... */
};
fn.name = bindingName
return fn;
```